lib/src/realm_class.dart

```diff
@@ -483,6 +483,40 @@ class Realm implements Finalizable {
       throw RealmError('Starting a write transaction on a frozen Realm is not allowed.');
     }
   }
+
+  /// Compacts a Realm file. A Realm file usually contains free/unused space.
+  ///
+  /// This method removes this free space and the file size is thereby reduced.
+  /// Objects within the Realm file are untouched.
+  /// Note: The file system should have free space for at least a copy of the Realm file. This method must not be called inside a transaction.
+  /// The Realm file is left untouched if any file operation fails.
+  static bool compact(Configuration config) {
+    late Configuration compactConfig;
+    if (config is LocalConfiguration) {
+      compactConfig = Configuration.local(config.schemaObjects.toList(),
+          schemaVersion: config.schemaVersion,
+          fifoFilesFallbackPath: config.fifoFilesFallbackPath,
+          path: config.path,
+          encryptionKey: config.encryptionKey,
+          disableFormatUpgrade: true,
+          isReadOnly: config.isReadOnly);
+    } else if (config is FlexibleSyncConfiguration || config is DisconnectedSyncConfiguration) {
+      compactConfig = Configuration.disconnectedSync(config.schemaObjects.toList(),
+          fifoFilesFallbackPath: config.fifoFilesFallbackPath, path: config.path, encryptionKey: config.encryptionKey);
+    } else if (config is InMemoryConfiguration) {
```

Comment:
Why can we even compact an in memory realm? Should this be an exception instead?

Reply:
I was contemplating that but decided to support it since core can do it

Reply:
Can you ping them and see if it actually does something? The thing is, in-memory Realms only live while somebody is referencing them, while compact requires that all realm instances are closed. So if I close the in-memory realm, it goes away and then compacting it will recreate it, compact the empty realm, then close it, thus destroying it.

Reply:
Why can we even compact an in memory realm?
>Free some memory perhaps

Reply:
I will throw an exception

Reply:
fixed